Libraries/CLI/src/cli.c

```diff
@@ -0,0 +1,265 @@
+ *
+ ******************************************************************************/
+
+/* -------------------------------------------------- */
+//                      INCLUDES
+/* -------------------------------------------------- */
+
+#include "cli.h"
+
+/* -------------------------------------------------- */
+//             FUNCTION PROTOTYPES
+/* -------------------------------------------------- */
+
+/** Checks if there is white space present in the given character pointer
+ *
+ * @param p
+ * 		  character pointer containing the line accumulator input
+ *
+ * @return 1 if white space is present
+ * 		   0 if no white space is present
+ */
+bool white_space_present(char *p);
+
+/** Checks if there is no white space present in the given character pointer
+ *
+ * @param p
+ * 		  character pointer containing the line accumulator input
+ *
+ * @return 1 if no white space is present
+ * 		   0 if white space is present
+ */
+bool white_space_not_present(char *p);
+
+void User_Prompt_Sequence(void);
+
+void Console_Backspace_Sequence(void);
+
+void Console_Cmd_Clear(void);
+
+void Clear_buffer(void);
+
+/* -------------------------------------------------- */
+//                      GLOBALS
+/* -------------------------------------------------- */
+
+/*! \name buf
+    \brief Buffer to store the characters of the command 
+*/
+char buf[256];
+int idx = 0;
+
+// Define a buffer to store commands
+#define MAX_COMMAND_LENGTH 256
+
+void line_accumlator(uint8_t user_char)
+{
+    switch (user_char) {
+    case BACKSPACE: {
+        // Handle Backspace and Delete
+        if (idx > 0) {
+            //Sequence to implement a backspace on the terminal
+            Console_Backspace_Sequence();
+            idx--;
+            buf[idx] = '\0';
+        }
+        break;
+    }
+    case ENTER: {
+        // Handle Enter or carriage return
+        MXC_UART_WriteCharacter(MXC_UART_GET_UART(CONSOLE_UART), NEW_LINE);
+        MXC_UART_WriteCharacter(MXC_UART_GET_UART(CONSOLE_UART), ENTER);
+        buf[idx++] = '\r';
+        buf[idx] = '\0';
+        idx = 0;
+        process_command(buf);
+        Clear_buffer();
+        break;
+    }
+
+    default: {
+        // Handle all other characters
+        if (idx < 255) {
+            buf[idx++] = user_char; //pushes characters into the buffer
+            MXC_UART_WriteCharacter(MXC_UART_GET_UART(CONSOLE_UART), user_char);
+        }
+        break;
+    }
+    }
+}
+
+void process_command(char *input)
+{
+    //Initialize p and end pointers
+    char *p = input;
+    char *end;
+
+    //Find end of string
+    for (end = input; *end != '\0'; end++) {}
+
+    //Initialize variables
+    bool in_token = false;
+    bool beginning_space;
+    if (*p == SPACE || *p == TAB)
+        beginning_space = true;
+    char *argv[10];
+    int argc = 0;
+    memset(argv, 0, sizeof(argv));
+
+    //Iterate over each character in input
+    for (p = input; p < end; p++) {
+        //If in a token
```

Comment:
Instead of iterating over all characters in the array, I would suggest using strchr() function, search for white_space, then create argv[][] array with all arguments given by the user. I think that could be computationally efficient.